compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -296,6 +299,20 @@ private void writeFunction(ByteBuf buf, BIRTypeWriter typeWriter, BIRInstruction
         buf.writeBytes(birbuf.nioBuffer().array(), 0, length);
     }
 
+    private void writeFunctionsGlobalVarDependency(ByteBuf buf, BIRNode.BIRFunction birFunction) {
+        List<Integer> globalVarBuf = new LinkedList<>();
+
+        for (BIRNode.BIRVariableDcl var : birFunction.dependentGlobalVars) {
+            if (var == null) {
+                continue;
+            }
+            globalVarBuf.add(addStringCPEntry(var.name.value));
+        }
+
+        buf.writeLong(globalVarBuf.size());
```

Comment:
Order is wrong here. you have to first write the size to the buffer and then write individual elements. And when reading, you read in the same order.

Reply:
I think I have read it wrong. This looks correct.